src/tests/suite/dfs_test.c

```diff
@@ -166,6 +167,16 @@ main(int argc, char **argv)
 		tests[new_idx] = '\0';
 	}
 
+	/** if writing XML, force all ranks other than rank 0 to use stdout to avoid conflicts */
+	cmocka_message_output = getenv("CMOCKA_MESSAGE_OUTPUT");
+	if (rank != 0 && cmocka_message_output && strcasecmp(cmocka_message_output, "xml") == 0) {
+		rc = d_setenv("CMOCKA_MESSAGE_OUTPUT", "stdout", 1);
+		if (rc) {
+			print_message("d_setenv() failed with %d\n", rc);
+			return -1;
+		}
+	}
+
```

Comment:
This comment explains the issue and solution:
https://daosio.atlassian.net/browse/DAOS-14594?focusedCommentId=123244

Reply:
In the latest run, the parallel test ran with 8 client processes per node
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13312/9/artifact/Functional%20Hardware%20Large/daos_test/dfs.py/job.log
And afterwards we see there is only one cmocka file from rank 0
```
2024-01-09 22:18:30,565 cmocka_utils     L0138 DEBUG| Remote /var/tmp/daos_testing/test_daos_dfs_parallel/cmocka directories:
2024-01-09 22:18:30,565 run_utils        L0410 DEBUG| Running on wolf-[205-208] with a 120 second timeout: ls -alR /var/tmp/daos_testing/test_daos_dfs_parallel/cmocka
2024-01-09 22:18:30,768 run_utils        L0266 DEBUG|   wolf-[206-208] (rc=0):
2024-01-09 22:18:30,768 run_utils        L0271 DEBUG|     /var/tmp/daos_testing/test_daos_dfs_parallel/cmocka:
2024-01-09 22:18:30,768 run_utils        L0271 DEBUG|     total 8
2024-01-09 22:18:30,768 run_utils        L0271 DEBUG|     drwxrwxr-x 2 jenkins jenkins 4096 Jan  9 22:18 .
2024-01-09 22:18:30,768 run_utils        L0271 DEBUG|     drwxrwxr-x 3 jenkins jenkins 4096 Jan  9 22:18 ..
2024-01-09 22:18:30,769 run_utils        L0266 DEBUG|   wolf-205 (rc=0):
2024-01-09 22:18:30,769 run_utils        L0271 DEBUG|     /var/tmp/daos_testing/test_daos_dfs_parallel/cmocka:
2024-01-09 22:18:30,769 run_utils        L0271 DEBUG|     total 16
2024-01-09 22:18:30,769 run_utils        L0271 DEBUG|     drwxrwxr-x 2 jenkins jenkins 4096 Jan  9 22:18 .
2024-01-09 22:18:30,769 run_utils        L0271 DEBUG|     drwxrwxr-x 3 jenkins jenkins 4096 Jan  9 22:18 ..
2024-01-09 22:18:30,769 run_utils        L0271 DEBUG|     -rw-rw-r-- 1 jenkins jenkins  754 Jan  9 22:18 DAOS_FileSystem_DFS_Parallel_cmocka_results.xml
2024-01-09 22:18:30,769 run_utils        L0271 DEBUG|     -rw-rw-r-- 1 jenkins jenkins  758 Jan  9 22:18 DAOS_FileSystem_DFS_Parallel_DTX_cmocka_results.xml
```